golang/cosmos/types/address_hooks.go

```diff
@@ -177,41 +177,67 @@ func extractBaseTransferData(transferData transfertypes.FungibleTokenPacketData,
 // If newPacket is not nil, it is populated with a new transfer packet whose
 // corresponding Sender or Receiver is replaced with the extracted base address.
 func ExtractBaseAddressFromPacket(cdc codec.Codec, packet ibcexported.PacketI, role AddressRole, newPacket *channeltypes.Packet) (string, error) {
+	var newDataP *[]byte
+	if newPacket != nil {
+		// Capture the data for the new packet.
+		newDataP = new([]byte)
+	}
+	target, err := ExtractBaseAddressFromData(cdc, packet.GetData(), role, newDataP)
+	if err != nil {
+		return target, err
+	}
+
+	if newPacket == nil {
+		return target, nil
+	}
+
+	// Create the new packet.
+	th := packet.GetTimeoutHeight()
+	*newPacket = channeltypes.NewPacket(
+		*newDataP, packet.GetSequence(),
+		packet.GetSourcePort(), packet.GetSourceChannel(),
+		packet.GetDestPort(), packet.GetDestChannel(),
+		clienttypes.NewHeight(th.GetRevisionNumber(), th.GetRevisionHeight()),
```

Comment:
What is the reasoning for sometimes using `NewHeight` like here, and sometimes using `MustParseHeight` ?